File: server/src/main/java/com/ahmadramadhan/mudahtitip/auth/dto/AuthResponse.java
package com.ahmadramadhan.mudahtitip.auth.dto;

import com.ahmadramadhan.mudahtitip.auth.UserRole;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * DTO for authentication response containing JWT token and user info.
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class AuthResponse {

    private String token;
    private String tokenType;
    private Long userId;
    private String name;
    private String email;
    private UserRole role;
    private Long shopId; // Only for SHOP_OWNER
}

File: client/lib/features/auth/data/models/auth_response.dart
import 'package:equatable/equatable.dart';
import 'user_role.dart';

/// Auth response from login/register endpoints.
class AuthResponse extends Equatable {
  final String token;
  final String tokenType;
  final int userId;
  final String name;
  final String email;
  final UserRole role;
  final int? shopId;

  const AuthResponse({
    required this.token,
    required this.tokenType,
    required this.userId,
    required this.name,
    required this.email,
    required this.role,
    this.shopId,
  });

  factory AuthResponse.fromJson(Map<String, dynamic> json) {
    return AuthResponse(
      token: json['token'] as String,
      tokenType: json['tokenType'] as String,
      userId: json['userId'] as int,
      name: json['name'] as String,
      email: json['email'] as String,
      role: UserRole.fromString(json['role'] as String),
      shopId: json['shopId'] as int?,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'token': token,
      'tokenType': tokenType,
      'userId': userId,
      'name': name,
      'email': email,
      'role': role.value,
      'shopId': shopId,
    };
  }

  @override
  List<Object?> get props => [
    token,
    tokenType,
    userId,
    name,
    email,
    role,
    shopId,
  ];
}
